documents-git/includes/providers/class-base-loader.php

```diff
@@ -264,43 +314,90 @@ protected function set_repo_details($url)
-        $this->set_repo_details($url);
-        list($raw_document, $response_code) = $this->get_document();
-
-        return array($raw_document, $response_code);
     }
 
     /**
      * Extracts the attributes from a shortcode. All attributes of all shortcodes are extracted,
      * but not necessarily passed, so they default to an empty string.
      *
-     * It also sets the class attributes "user", "token" and "limit" from config.json or shortcode attribute.
+     * It also sets the class attributes "user", "token", "cache_ttl" and "limit" from config.json or shortcode attribute.
      *
      * @param $attrs array Attributes of the shortcode
      * @return string parsed url
      */
-    private function extract_attributes($attrs) {
-
+    private function extract_attributes(array $attrs)
+    {
         $attrs = array_change_key_case((array)$attrs, CASE_LOWER);
         extract(shortcode_atts(array(
                 'url' => "",
                 'user' => "",
                 'token' => "",
-                'limit' => ""
+                'limit' => "",
+                'cache_ttl' => "",
+                'cache_strategy' => "",
             ), $attrs
             )
         );
 
         $this->user = ($user === '') ? (MARKDOWNGIT_CONFIG[static::$PROVIDER]["user"]) : ($user);
         $this->token = ($token === '') ? (MARKDOWNGIT_CONFIG[static::$PROVIDER]["token"]) : ($token);
         $this->limit = ($limit === '') ? (MARKDOWNGIT_CONFIG["limit"]) : ($limit);
+        $this->cache_ttl = ($cache_ttl === '') ? (MARKDOWNGIT_CONFIG["cache_ttl"]) : ($cache_ttl);
+        $this->cache_strategy = ($cache_strategy === '') ? (MARKDOWNGIT_CONFIG["cache_strategy"]) : ($cache_strategy);
 
         return $url;
     }
-}
\ No newline at end of file
+
+    /**
+     * Get cached content when cache is enabled.
+     *
+     * @param string $url cache key to be serialized.
+     * @param string $group group where content was stored.
+     * @return mixed
+     *
+     * @since 1.1.0
+     */
+    private function get_cached_content(string $url, string $group)
+    {
+        return get_transient($this->get_cache_key($url, $group));
+    }
+
+    /**
+     * Caches content using a cache key
+     *
+     * @param string $url cache key.
+     * @param string $group group where content was stored.
+     * @param mixed $content content to cache.
+     *
+     * @since 1.1.0
+     */
+    private function set_content_cache(string $url, string $group, $content)
+    {
+        set_transient($this->get_cache_key($url, $group), $content, (int) $this->cache_ttl);
+    }
+
+    /**
+     * Constructs the cache key from the attributes.
+     *
+     * @param string $url The URL of the document
+     * @param string $group The cache group
+     *
+     * @since 1.1.0
+     */
+    private function get_cache_key(string $url, string $group)
+    {
+        return md5(self::$NAMESPACE . $group . $url . strval($this->cache_ttl) . strval($this->limit));
+    }
```

Comment:
Probably makes more sense to base the key on the class attributes which don't only depend on the config.json but also the shortcode attributes. Had to rearrange a little bit to allow for that.